integrationTests/realcomponents/processorRunner.go

```diff
@@ -124,8 +139,9 @@ func (pr *ProcessorRunner) createCoreComponents(tb testing.TB) {
 		NodesFilename:            pr.Config.ConfigurationPathsHolder.Nodes,
 		WorkingDirectory:         pr.Config.FlagsConfig.WorkingDir,
 		ChanStopNodeProcess:      make(chan endProcess.ArgEndProcess),
-		GenesisNodesSetupFactory: sharding.NewGenesisNodesSetupFactory(),
-		RatingsDataFactory:       rating.NewRatingsDataFactory(),
+		GenesisNodesSetupFactory: pr.RunTypeCoreComponents.GenesisNodesSetupFactoryCreator(),
```

Comment:
Do not pass all these subcomps one by one.
Just pass `RunTypeCoreComponents`, as it includes anything you need anyway